### PR TITLE
Add authorization rules

### PR DESCRIPTION
**Pull request by lordksix**
In this PR, the changes are as follows:
- [x] Install CanCanCan in your project.
- [x] Add a role column to the users table. Remember to use a migration for this.
- [x]  A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
  - [x] For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
- [x] A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
  - [x] For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

**Review by TedLivist — COMMENTED** [review]
## STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @lordksix and @lenkon,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### The third team member needs to contribute to this project. Kindly ensure that happens.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  def initialize(user)
+    user ||= User.new
+
+    can %i[index show], User
+    can %i[index new create show], Post
+    can %i[new create], Comment
+    can %i[update destroy], Post, { author_id: user.id }
+    can %i[update destroy], Comment, { author_id: user.id }
+    can :manage, Post if user.is? :admin
+    can :manage, Comment if user.is? :admin
```
Comment (VICTOR-CHUKWU) [inline app/models/ability.rb:13]: - [x] you can actually refactor this to `   can :manage, :all if user.is? :admin` :+1:

File: app/models/user.rb [hunk 2]
```diff
@@ -4,13 +4,17 @@ class User < ApplicationRecord
...
+  has_many :posts, foreign_key: 'author_id', dependent: :destroy
+  has_many :comments, foreign_key: 'author_id', dependent: :destroy
+  has_many :likes, foreign_key: 'author_id', dependent: :destroy
 
   validates :name, presence: true
   validates :post_counter, numericality: { greater_than_or_equal_to: 0 }
 
+  def is?(requested_role)
+    role == requested_role.to_s
+  end
```
Comment (VICTOR-CHUKWU) [inline app/models/user.rb:16]: - [x] nice implementation :clap:

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @lordksix , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- [ ] admin role defined
- [ ] user role defined
- [ ] cancancan ability installed and used

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**